spec/requests/trashbin_api_spec.rb

```diff
@@ -64,83 +64,178 @@
 
   describe 'PUT /trashbin/{object_kind}/{object_id}/restore' do
     let(:url) { "/api/v1/trashbin/#{resource_kind}/#{resource_id}/restore" }
-    subject { put(url, params: payload.to_json, headers: headers) }
     let(:called_action) { 'PUT' }
-    let(:parent_kind) { parent_folder.kind }
-    let(:parent_id) { parent_folder.id }
-    let(:resource) { trashed_resource }
-    let(:resource_class) { DataFile }
-    let(:resource_serializer) { DataFileSerializer }
-    let(:resource_kind) { trashed_resource.kind }
-    let(:resource_id) { trashed_resource.id }
 
-    let(:payload) {{
-      parent: {
-        kind: parent_kind,
-        id: parent_id
+    context 'container object' do
+      let(:parent_kind) { parent_folder.kind }
+      let(:parent_id) { parent_folder.id }
+      let(:resource) {
+        trashed_resource
       }
-    }}
+      let(:resource_class) { DataFile }
+      let(:resource_serializer) { DataFileSerializer }
+      let(:resource_kind) { trashed_resource.kind }
+      let(:resource_id) { trashed_resource.id }
+      before do
+        resource.move_to_trashbin
+        resource.save
+      end
 
-    it_behaves_like 'an identified resource' do
-      let(:resource_id) { "doesNotExist" }
-    end
+      context 'without payload' do
+        subject { put(url, headers: headers) }
 
-    it_behaves_like 'an identified resource' do
-      let(:resource_id) { untrashed_resource.id }
-    end
+        context 'resource in root of the project' do
+          it_behaves_like 'an identified resource' do
+            let(:resource_id) { "doesNotExist" }
+          end
 
-    it_behaves_like 'an identified resource' do
-      let(:parent_id) { "doesNotExist" }
-      let(:resource_class) { Folder }
-    end
+          it_behaves_like 'an identified resource' do
+            let(:resource_id) { untrashed_resource.id }
+          end
 
-    it_behaves_like 'a client error' do
-      let(:expected_response) { 404 }
-      let(:expected_reason) { "dds-folder #{parent_folder.id} is deleted, and cannot restore children." }
-      let(:expected_suggestion) { "Restore #{parent_folder.kind} #{parent_folder.id}." }
-      before do
-        parent_folder.update_columns(is_deleted: true)
-      end
-    end
+          context 'that is deleted' do
+            before do
+              project.update_columns(is_deleted: true)
+            end
+            it_behaves_like 'a client error' do
+              let(:expected_response) { 404 }
+              let(:expected_reason) { "dds-project #{project.id} is permenantly deleted, and cannot restore children." }
+              let(:expected_suggestion) { "Restore to a different project." }
```

Comment:
I'm not asking for a change here, but is this even possible? I'm under the impression that when a project is deleted, it purges its children. Maybe this can happen with projects/children that are currently `is_deleted` in dds?

Reply:
It may just be defensive